### ISSUE TITLE
how can I use testament to test all tests?

### ISSUE THREAD

**Issue by ringabout** [nim-lang/Nim#15020]
`testament` is supposed to be a standalone tool. But `testament all` is an invalid command which makes it a bit unclear to use `testament`. I just use `testament cat /` as a workaround which works but is not documented well. I don't know other means.

https://github.com/planety/prologue/blob/973c8db9332448b57b74e8e40ad7e2d284b962d6/prologue.nimble#L23

https://github.com/iocrate/netkit/blob/3ac9edd4033f24c552e8963f3e14e5b4eb92d405/netkit.nimble#L13

```
>>> D:\QQPCmgr\Desktop\iNim\cookbook>testament all
progress[all]: 0/5 starting: cat: debugger
C:\Users\blue\.choosenim\toolchains\nim-#devel\bin\testament.exe --nim:C:\Users\blue\.nimble\bin\nim.exe pcat debugger       
progress[all]: 1/5 starting: cat: examples
C:\Users\blue\.choosenim\toolchains\nim-#devel\bin\testament.exe --nim:C:\Users\blue\.nimble\bin\nim.exe pcat examples       
progress[all]: 2/5 starting: cat: lib
C:\Users\blue\.choosenim\toolchains\nim-#devel\bin\testament.exe --nim:C:\Users\blue\.nimble\bin\nim.exe pcat lib
progress[all]: 3/5 starting: cat: ic
C:\Users\blue\.choosenim\toolchains\nim-#devel\bin\testament.exe --nim:C:\Users\blue\.nimble\bin\nim.exe pcat ic
progress[all]: 4/5 starting: cat: megatest
C:\Users\blue\.choosenim\toolchains\nim-#devel\bin\testament.exe --nim:C:\Users\blue\.nimble\bin\nim.exe pcat megatest       
joinable specs: 0
io.nim(782)              writeFile
Error: unhandled exception: cannot open: tests\megatest.nim [IOError]
```

**Comment by ringabout** [comment]
I think `testament all` should run all tests in `test` directory for external projects. We should choose other commands to test the entire Nim compiler like `testament nimc` or we can document `testament cat /` as a start.

**Comment by timotheecour** [comment]
`testament cat /` seems wrong for starters; if that works, that's a bug IMO; `/` should point to the root, not PWD (`.`)

**Comment by ringabout** [comment]
As far as I'm concerned. `testament cat .` also works. Anyway there should be at least one way to run all tests and this way should be documented.

**Comment by alaviss** [comment]
Works for my (WIP) library. What version of nim are you on?

```
$ testament all
progress[all]: 0/6 starting: cat: pipes
/usr/lib/nim/bin/testament --nim:/usr/lib/nim/bin/nim pcat pipes
progress[all]: 1/6 starting: cat: debugger
/usr/lib/nim/bin/testament --nim:/usr/lib/nim/bin/nim pcat debugger
progress[all]: 2/6 starting: cat: examples
/usr/lib/nim/bin/testament --nim:/usr/lib/nim/bin/nim pcat examples
progress[all]: 3/6 starting: cat: lib
/usr/lib/nim/bin/testament --nim:/usr/lib/nim/bin/nim pcat lib
progress[all]: 4/6 starting: cat: ic
/usr/lib/nim/bin/testament --nim:/usr/lib/nim/bin/nim pcat ic
progress[all]: 5/6 starting: cat: megatest
/usr/lib/nim/bin/testament --nim:/usr/lib/nim/bin/nim pcat megatest
joinable specs: 0
megatest output OK
PASS: tests/pipes/tasyncreadwrite.nim C                            ( 0.93 sec)
PASS: tests/pipes/tdestroy.nim C                                   ( 0.77 sec)
PASS: tests/pipes/tdestroy.nim C                                   ( 0.82 sec)
PASS: tests/pipes/treadwrite.nim C                                 ( 0.79 sec)
PASS: tests/pipes/treadwrite.nim C                                 ( 0.85 sec)
SKIP: tests/pipes/twindows.nim C
```

**Comment by xflywind** [comment]
My Nim version is Nim 1.3.5(2020-7-17) and I'm in windows.

**Comment by ringabout** [comment]
Oh sorry, it's fixed since 1.2.2.
